lamindb/_artifact.py

```diff
@@ -615,16 +617,29 @@ def __init__(artifact: Artifact, *args, **kwargs):
     else:
         kwargs = kwargs_or_artifact
 
+    if data is not None:
+        artifact._local_filepath = privates["local_filepath"]
+        artifact._cloud_filepath = privates["cloud_filepath"]
+        artifact._memory_rep = privates["memory_rep"]
+        artifact._to_store = not privates["check_path_in_storage"]
+
+    if is_automanaged_path and _is_internal_call:
+        kwargs["_key_is_virtual"] = True
+        if AUTO_KEY_PREFIX in kwargs["key"]:
+            kwargs["key"] = None
+        provisional_uid = artifact.path.stem
```

Comment:
I don't think that works at this point.

Reply:
I mean `artifact.path`.

Reply:
Yes, I realized meanwhile...

This whole PR got much more complicated than I wanted 😞